src/core/helpers.js

```diff
@@ -1252,39 +1252,61 @@ wb.escapeAttribute = function( str ) {
 
 /*
 * Find most common Personal Identifiable Information (PII) in a string and return either the cleaned string either true/false
-* @param {string} str
-* @param {boolean} toClean
+* @param{string} str (required)- the content that needs to be verified
+* @param {boolean} toClean - if true will scrub the content
+* @param {object} the 2nd param can also be an  object having the following properties:*
+* 	{boolean} bMarker (optional) - if true, will replace the scrubbed values with string "[REDACTED/CAVIARDÉ]"
+* 	{string} caseOnly (optional) - this param should match a key name from the "regEx" object. If present, it will search and replace only the values corresponding that pattern
+* 	{regex} customCase (optional) - this param is a regex. It will search and replace the values corresponding that pattern
+
 * @return {string | true | false}
 * @example
 * wb.findPotentialPII( "email:test@test.com, phone:123 123 1234", true )
 * returns "email:, phone:",
+*
 * wb.findPotentialPII( "email:test@test.com, phone:123 123 1234", false )
 * returns true
+*
+* wb.findPotentialPII( "email:test@test.com, phone:123 123 1234", { "bMarker":true, "caseOnly":"email" } )
+* returns "email:[REDACTED/CAVIARDÉ], phone:123 123 1234"
+*
+* wb.findPotentialPII( "email:test@test.com, phone:123 123 1234, numéro de cas 12345678", { "bMarker":true, "customCase":/\b(?:case[\s-]?number[\s\-\\.]?(?:\d{5,10}))|(?:numéro[\s-]?de[\s-]?cas[\s\-\\.]?(?:\d{5,10}))/ig })
+* returns "email:[REDACTED/CAVIARDÉ], phone:[REDACTED/CAVIARDÉ], [REDACTED/CAVIARDÉ]"
 */
-wb.findPotentialPII = function( str, toClean ) {
-
-	if ( typeof str  !== "string" ) {
+wb.findPotentialPII = function( str, opts ) {
+	if ( str && typeof str  !== "string" ) {
 		return false;
 	}
-	var regEx = [
-			/\d(?:[\s\-\\.\\/]?\d){7,}(?!\d)/ig, //8digits or more pattern
-			/\b[A-Za-z]{2}[\s\\.-]*?\d{6}\b/ig, //canadian nr passport pattern
-			/\b(?:[a-zA-Z0-9_\-\\.]+)(?:@|%40)(?:[a-zA-Z0-9_\-\\.]+)\.(?:[a-zA-Z]{2,5})\b/ig, //email pattern
-			/\b[A-Za-z]\d[A-Za-z][ -]?\d[A-Za-z]\d\b/ig, //postal code pattern
-			/\b(?:(username|user)[:=][a-zA-Z0-9_\-\\.]+)\b/ig,
-			/\b(?:(password|pass)[:=][^\s#&]+)\b/ig
-		],
-		isFound = false;
+	var regEx = {
+			"digits": /\d(?:[\s\-\\.\\/]?\d){8,}(?!\d)/ig, //9digits or more pattern
+			"passport": /\b[A-Za-z]{2}[\s\\.-]*?\d{6}\b/ig, //canadian nr passport pattern
+			"email": /\b(?:[a-zA-Z0-9_\-\\.]+)(?:@|%40)(?:[a-zA-Z0-9_\-\\.]+)\.(?:[a-zA-Z]{2,5})\b/ig, //email pattern
+			"postal-code": /\b[A-Za-z]\d[A-Za-z][ -]?\d[A-Za-z]\d\b/ig, //postal code pattern
+			"username": /\b(?:(username|user)[:=][a-zA-Z0-9_\-\\.]+)\b/ig,
+			"password": /\b(?:(password|pass)[:=][^\s#&]+)\b/ig
+		},
+		isFound = false,
+		txMarker =  "[REDACTED/CAVIARDÉ]",
```

Comment:
We will need to localize it. We would need to add an entry in the i18n spreadsheet

Check example that use "i18nText".

Or we can just fall back on the FullBlock character by default when no maker is provided.

The current practice is to replace every character by the marker, but your proposal seem is to replace the whole redacted text with a single phrase